Engine/Fuego/Windows/WindowWin.h

```diff
@@ -25,9 +27,16 @@ class WindowWin : public Window
     virtual void SetVSync(bool enabled) override;
     virtual bool IsVSync() const override;
 
+    LRESULT WindowProc(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam);
+    static std::unordered_map<HWND, WindowWin*> hwndMap;
+
    private:
     void Init(const WindowProps& props, EventQueue& eventQueue);
     void Shutdown();
+    static LRESULT CALLBACK WindowProcStatic(HWND hWnd, UINT uMsg,
+                                             WPARAM wParam, LPARAM lParam);
+
+    EventQueue& m_EventQueue;
```

Comment:
Why do we need a reference here?

Reply:
Cause WindowProc func has to be static and at the same time I need access to WindowQueue members, so I map Static func to window api  and then I cann member func